binding.gyp

```diff
@@ -2,15 +2,26 @@
   "targets": [
     {
       "target_name": "fontmanager",
-      "sources": [ "src/FontManager.cc" ],
-      "include_dirs" : [
-        "<!(node -e \"require('nan')\")"
+      "sources": [
+        "src/FontDescriptor.cc",
+        "src/FontManager.cc"
+      ],
+      "include_dirs": [
+        "<!(node -p \"require('node-addon-api').include_dir\")"
+      ],
+      "defines": [
+        "NAPI_DISABLE_CPP_EXCEPTIONS",
+        "NODE_ADDON_API_ENABLE_MAYBE"
```

Comment:
- Why disable exceptions? I'm not against it if that's the easiest way to convert to napi, but probably it should be commented out in the Windows code that throws.

- On Windows, I tested defining `NAPI_CPP_EXCEPTIONS` instead of `NAPI_DISABLE_CPP_EXCEPTIONS` and `NODE_ADDON_API_ENABLE_MAYBE`, and it doesn't compile. `npm install` shows the error message: `c:\users\alex\repos\axosoft\font-scanner\src\fontdescriptor.cc(24): error C2039: 'UnwrapTo': is not a member of 'Napi::Value
' [C:\Users\alex\Repos\Axosoft\font-scanner\build\fontmanager.vcxproj]`

  It seems the reason is that `UnwrapTo` is a member of `Napi::Maybe`, not of `Napi::Value`.
  Would something like this work in this case?
  ```c++
  Napi::MaybeLocal<v8::Value> maybeValue = obj.Get(Napi::New(env, name));

  if (maybeValue.IsEmpty()) {
    return NULL;
  }
  ```

Reply:
the exceptions in this context actually refer to how the napi converts JS types:
 - With exceptions enabled it will convert types directly and throw an exception if it doesn't convert. since there's no manual  checking, the Maybe<> Type isn't used and actually isn't even defined.
 - With exceptions disabled, the result of conversions have to be checked manually, hence the existence of the Maybe<> type.

so functionally there's not much difference, I went with disabling exceptions since it mostly maintains the structure of the existing code and because I'm not sure how c++ exceptions affect node and whether we'd need to catch and convert them manually.

Reply:
Thanks for the explanation! 👍